docs/user/guides/hosting-integrations.rst

```diff
@@ -0,0 +1,97 @@
+Hosting Integrations
+====================
+
+Read the Docs performs some actions by default when building documentation for its users to.
+These actions integrate the documentation project with multiple hosting features Read the Docs offers.
+However, when using ``build.commands`` configuration (see :doc:`build-customization`),
+the build process is overriden completely and it's the user responsibility to integrate these features.
+This page explains how to manually integrate these features when Read the Docs cannot do it automatically.
+
+These are all the hosting features that Read the Docs provides by default:
+
+`Flyout menu`_
+   Menu displaying on all documentation pages containing the list of active versions,
+   downloadable formats and other useful links.
+   Read more at :doc:`flyout-menu`.
+
+`Search index`_
+   Smart search index powered by Elasticsearch able to differentiate between regular text,
+   function names in code, and search in all subprojects, among others.
+   Read more at :doc:`server-side-search`.
+
+`Search as you type`_
+   Level up your search by providing immediate feedback while the user is typing.
+   Read more at :doc:`advanced-search.html#enable-search-as-you-type-in-your-documentation`.
+
+`Version warning`_
+   Banner displayed on old versions when there is a newer version available,
+   communicating readers they may be careful since they are reading an older version of the documentation.
+   Read more at :doc:`versions#version-warning`.
+
+`External (pull request) version warning`_
+   Benner displayed on versions of the documentation built from an external version (pull request),
+   warning readers this is not the final version of this documentation.
+   Read more at :doc:`pull-requests`.
+
+.. TODO: write a new page explaining this and link it from here.
```

Comment:
I'm pretty sure users won't care about integrating this, but the page explaining that we do it with some public metabase charts could be cool.

Reply:
Yeah, I'm not expecting final users to integrate this, but probably doctools/themes/etc. I think it's fine to have it documented and let people decide whether or not they want to support us by dumping this data. The more data they dump, the better support we can offer to them. So, it's a win-win in the end.